api/src/main/java/org/openmrs/validator/ConceptValidator.java

```diff
@@ -113,6 +115,9 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 			Set<String> validNamesFoundInLocale = new HashSet<>();
 			Collection<ConceptName> namesInLocale = conceptToValidate.getNames(conceptNameLocale);
 			for (ConceptName nameInLocale : namesInLocale) {
+				if (conceptToValidate.getRetired()) {
+                    continue;
+                }
```

Comment:
It seems like it would be more efficient to move this if check before the outer loop, since if a concept is retired, we don't need to validate any of it's names.